ironfish/src/migrations/data/033-multisig-keys-identity/new/accountValue.ts

```diff
@@ -0,0 +1,229 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/. */
+import { KEY_LENGTH, PUBLIC_ADDRESS_LENGTH, xchacha20poly1305 } from '@ironfish/rust-nodejs'
+import bufio from 'bufio'
+import { IDatabaseEncoding } from '../../../../storage'
+import { HeadValue, NullableHeadValueEncoding } from './headValue'
+import { MultisigKeys } from './interfaces/multisigKeys'
+import { MultisigKeysEncoding } from './multisigKeys'
+
+export const VIEW_KEY_LENGTH = 64
+const VERSION_LENGTH = 2
+
+export type EncryptedAccountValue = {
+  encrypted: true
+  salt: Buffer
+  nonce: Buffer
+  data: Buffer
+}
+
+export type DecryptedAccountValue = {
+  encrypted: false
+  version: number
+  id: string
+  name: string
+  spendingKey: string | null
+  viewKey: string
+  incomingViewKey: string
+  outgoingViewKey: string
+  publicAddress: string
+  createdAt: HeadValue | null
+  scanningEnabled: boolean
+  multisigKeys?: MultisigKeys
+  proofAuthorizingKey: string | null
+}
+
+export type AccountValue = EncryptedAccountValue | DecryptedAccountValue
+export class AccountValueEncoding implements IDatabaseEncoding<AccountValue> {
+  serialize(value: AccountValue): Buffer {
+    if (value.encrypted) {
+      return this.serializeEncrypted(value)
+    } else {
+      return this.serializeDecrypted(value)
+    }
+  }
+
```

Comment:
Do we need to do this for every account migration? Have separate encrypted/decrypted serde? That is brutal, but I think I remember you talking about this and finding it was required.

Reply:
Yeah we have separate serde in the database now, so we will always need to handle both for migrations 😭